### PR TITLE
client-go: Move custom <resource> struct methods to <resource>_expansion files

### PR DESCRIPTION
**Pull request by fossedihelm**
##### Before this PR:
Our `vm`,  `vmi`, `kv`, `vmirs` structs respectively embed generated
`VirtualMachineInterface`, `VirtualMachineInstanceInterface`,
`KubeVirtInterface` and  `VirtualMachineInstanceResplicasetInterface`.
The latter define the custom methods but do not
implement them, which are only defined in the custom structs.

##### After this PR:
We can move the expanded methods implementations
in the `virtualmachine_expansion`, `virtualmachineinstance_expansion`,
`kubevirt_expansion`, `virtualmachineinstancereplicaset_expansion` files so that our custom structs
can simply inherit them.

Also, switched to using full `AbsPath` with fmt.Sprintf()
to request composition with `Namespace()`, `Resource()`,
`Name()` and `SubResource()`.

### Special notes for your reviewer
N.B. methods that require clientConfig, like `PortForward`, remain implemented only 
in our custom structs because we still cannot implement them in the
expansion files, due to the lack of the clientConfig.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
@vasiliy-ul Can you have a look? Thanks

File: staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/virtualmachineinstance_expansion.go [hunk 1]
```diff
@@ -55,100 +62,306 @@ type VirtualMachineInstanceExpansion interface {
...
+		},
+	}
+
+	body, err := json.Marshal(freezeUnfreezeTimeout)
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
```
Comment (vasiliy-ul) [inline staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/virtualmachineinstance_expansion.go:157]: The second err check is redundant. Probably a leftover.

**Review by vasiliy-ul — COMMENTED** [review]
Looks good. Just one small issue.

File: staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/virtualmachineinstance_expansion.go (hunk 1 above)
Comment (fossedihelm) [reply staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/virtualmachineinstance_expansion.go]: Changed, PTAL Thanks

**Review by vasiliy-ul — APPROVED** [review]

**Pull request merged**